src/libPMacc/include/nvidia/atomic.hpp

```diff
@@ -51,22 +52,28 @@ namespace nvidia
  * nvidia parallel forall example
  * http://devblogs.nvidia.com/parallelforall/cuda-pro-tip-optimized-filtering-warp-aggregated-atomics/
  */
+template<typename T>
 DINLINE
-int atomicAllInc(int *ptr)
+T atomicAllInc(T *ptr)
 {
 #if (__CUDA_ARCH__ >= 300)
```

Comment:
even that could become a template parameter and could be used for a specialization